scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -44,149 +49,233 @@ set.seed(12345)
+
+```{r plot CD3, live = TRUE}
+# plot CD3 expression
+scater::plotUMAP(sce, colour_by = "CD3")
 ```
 
-CD4+ T-cells
+It appears that we have a number of potential T cells down in the lower left!
 
-```{r}
+Let's look at a couple of other markers to try to break those up more specifically.
+
+Two other markers of relevance to the T cells are `CD4` and `CD8`.
+The `CD4` complex is present in helper T cells (hence their other common name, CD4+ T cells).
+By contrast, the `CD8` complex is found on killer T cells (CD8+ cells).
+
+Let's plot the ADT results for those two markers as well below:
+
+```{r plot CD4, live=TRUE}
+# plot CD4 marker
 scater::plotUMAP(sce, 
                  colour_by = "CD4")
 ```
-CD8+ T-cells
 
-```{r}
+```{r plot CD8, live=TRUE}
+# plot CD8 marker
 scater::plotUMAP(sce, 
                  colour_by = "CD8")
 ```
 
 
-So one thing we could do is construct a set of rules based on ADT tags, similar to how you might set gating rules for cell sorting.
+### Rule-based classification
 
+When we have markers like these, with clear and established cell types, we might want to start our classification by working with these markers alone.
+Such classification might be considered as analogous to a cell-sorter assay, where we would set up some rules to look at a few markers for each cell and use those to assign a cell type classification.
 
-What thresholds should we use?
-If this is a useful marker, we should see a bimodal plot of the marker density.
+To do this, we will need to make some decisions, like what thresholds should we use to determine whether a cell is or is not expressing a particular marker. 
+In general, if a marker is going to be useful for this type of assignment, we would like to see a bimodal plot for the marker in question that can be used to separate the population into two groups of cells based on the level of that marker.
 
-We'll do a bit of wrangling to pull out the ADT data, transpose it so each row is a cell,
-then convert it into a data frame for plotting. 
-(We have to first convert to a "regular" R matrix or R won't know how to convert it to a data frame)
+To look at the expression levels of each of the marker genes we are interested in, we will pull out the expression values for the markers from the SCE object and do some data wrangling. 
 
-```{r}
-# convert to a data frame
+The data we are interested is the normalized count data for the ADT tags, which is stored in the `logcounts` assay.
+If you recall, this matrix is stored with the columns as cells and rows as markers, but we really want it the other way, so we will first transpose the data, then convert it to a data frame for our next steps.
+Because the SCE object stores the main data matrices in a specialized format, we have to do one extra step convert it first to a "regular" R matrix or R won't know how to convert it to a data frame.
```

Comment:
"main" isn't my favorite here given `main` vs `altExp` and we're talking about assays here.
I suggested "optimized for memory" as well to contextualize what this specialized format is. I can't decide if this is actually a "non-technical" term or if it's still too loaded though.

```suggestion
Because the SCE object stores the assays matrices in a specialized format optimized for memory, we have to do one extra step convert it first to a "regular" R matrix or R won't know how to convert it to a data frame.
```

Reply:
Yeah, I didn't want to get into this, as the real truth is that ADT matrices may or may not be stored as sparse (they are not sparse!), so I think I'm going to stick with the hand waving...